openedx/core/djangoapps/xblock/runtime/learning_core_runtime.py

```diff
@@ -300,19 +337,90 @@ def _get_component_from_usage_key(self, usage_key):
 
         return component
 
-    def _lookup_asset_url(self, block: XBlock, asset_path: str) -> str | None:  # pylint: disable=unused-argument
+    def _get_component_version_from_block(self, block):
+        usage_key = block.usage_key
+        component = self._get_component_from_usage_key(usage_key)
+
+        block_version = block._runtime_requested_version  # pylint: disable=protected-access
+        if block_version == LatestVersion.DRAFT:
+            component_version = component.versioning.draft
+        elif block_version == LatestVersion.PUBLISHED:
+            component_version = component.versioning.published
+        else:
+            component_version = component.versioning.version_num(block_version)
+
+        return component_version
+
+    def _lookup_asset_url(self, block: XBlock, asset_path: str) -> str | None:
         """
         Return an absolute URL for the specified static asset file that may
         belong to this XBlock.
 
-        e.g. if the XBlock settings have a field value like "/static/foo.png"
-        then this method will be called with asset_path="foo.png" and should
-        return a URL like https://cdn.none/xblock/f843u89789/static/foo.png
+        e.g. if the XBlock settings have a field value like "/static/test.png"
+        then this method will be called with asset_path="test.png" and should
+        return a URL like:
+
+          http://studio.local.edly.io:8001/library_assets/cd31871e-a342-4c3f-ba2f-a661bf630996/static/test.png
```

Comment:
Nit: we're moving toward `local.openedx.io` so I'd recommend just using that in all docs going forward. e.g. the new [`npm run dev` command](https://github.com/openedx/frontend-app-authoring?tab=readme-ov-file#startup) for running the MFEs in dev mode on your host (in Authoring & Learning MFEs only) uses that domain exclusively.